libs/pages/services/src/lib/ui/page-settings-general/page-settings-general.spec.tsx

```diff
@@ -0,0 +1,36 @@
+import { useForm } from 'react-hook-form'
+import { clusterFactoryMock } from '@console/domains/organization'
+import { render } from '__tests__/utils/setup-jest'
+import PageSettingsGeneral, { PageSettingsGeneralProps } from './page-settings-general'
+
+describe('PageSettingsGeneral', () => {
+  let props: PageSettingsGeneralProps
+  let Wrapper: React.FC
+
+  beforeEach(() => {
+    props = {
+      onSubmit: jest.fn(),
+      control: null as any,
+      clusters: clusterFactoryMock(2),
+    }
+
+    Wrapper = () => {
```

Comment:
Last time I was struggling with testing the forms indeed.
I found this very short notation that I liked. If you don't use the control
```
const WrapperForm = ({ children }) => {
  const methods = useForm()
  return <FormProvider {...methods}>{children}</FormProvider>
}

describe('CrudEnvironmentVariableModalFeature', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <WrapperForm>
        <CrudEnvironmentVariableModalFeature {...props} />
      </WrapperForm>
    )
    expect(baseElement).toBeTruthy()
  })
})
```

But what I wanted to point out here is that there is no need to pass control or any information linked to the form. You can directly use the context instead with:

`const { control, formState } = useFormContext()`

Which is pretty handy, simplify testing and simplify component declarations